contracts/test/StakeTable.t.sol

```diff
@@ -0,0 +1,272 @@
+    LightClient public lightClientContract;
+    uint256 constant INITIAL_BALANCE = 1_000;
+    address exampleTokenCreator;
+
+    function genClientWallet(address sender)
+        private
+        returns (BN254.G2Point memory, EdOnBN254.EdOnBN254Point memory, BN254.G1Point memory)
+    {
+        // Generate a BLS signature and other values using rust code
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "gen-client-wallet";
+        cmds[2] = vm.toString(sender);
+
+        bytes memory result = vm.ffi(cmds);
+        (
+            uint256 blsSigX,
+            uint256 blsSigY,
+            uint256 blsVKx0,
+            uint256 blsVKx1,
+            uint256 blsVKy0,
+            uint256 blsVKy1,
+            uint256 schnorrVKx,
+            uint256 schnorrVKy
+        ) = abi.decode(
+            result, (uint256, uint256, uint256, uint256, uint256, uint256, uint256, uint256)
+        );
+
+        return (
+            BN254.G2Point(blsVKx1, blsVKx0, blsVKy1, blsVKy0), // blsVK
+            EdOnBN254.EdOnBN254Point(schnorrVKx, schnorrVKy), // schnorrVK
+            BN254.G1Point(blsSigX, blsSigY) // sig
+        );
+    }
+
+    function setUp() public {
+        exampleTokenCreator = makeAddr("tokenCreator");
+        vm.prank(exampleTokenCreator);
+        token = new ExampleToken(INITIAL_BALANCE);
+
+        LightClient.LightClientState memory genesis = LightClient.LightClientState({
+            viewNum: 0,
+            blockHeight: 0,
+            blockCommRoot: 0,
+            feeLedgerComm: 0,
+            stakeTableBlsKeyComm: 0,
+            stakeTableSchnorrKeyComm: 0,
+            stakeTableAmountComm: 0,
+            threshold: 0
+        });
+        lightClientContract = new LightClient(genesis,10);
+        address lightClientAddress = address(lightClientContract);
+        stakeTable = new S(address(token),lightClientAddress);
+    }
+
+    function test_RevertWhen_UsingRestakeToken() external {
+        uint64 depositAmount = 10;
+        uint64 validUntilEpoch = 5;
+
+        (
+            BN254.G2Point memory blsVK,
+            EdOnBN254.EdOnBN254Point memory schnorrVK,
+            BN254.G1Point memory sig
+        ) = genClientWallet(exampleTokenCreator);
+
+        // Throw "Restaking not implemented" error
+        vm.expectRevert(S.RestakingNotImplemented.selector);
+        stakeTable.register(
+            blsVK,
+            schnorrVK,
+            depositAmount,
+            AbstractStakeTable.StakeType.Restake,
+            sig,
+            validUntilEpoch
+        );
+    }
+
+    function test_RevertWhen_InvalidBLSSig() external {
+        uint64 depositAmount = 10;
+        uint64 validUntilEpoch = 5;
+
+        (BN254.G2Point memory blsVK, EdOnBN254.EdOnBN254Point memory schnorrVK,) =
+            genClientWallet(exampleTokenCreator);
+
+        // Failed signature verification
+        BN254.G1Point memory badSig = BN254.P1();
```

Comment:
I would fuzz this by passing in a random scalar, then use `BN254.scalarMul` to get the random point as the badSig.